lib/fika/types.ex

```diff
@@ -2,15 +2,4 @@ defmodule Fika.Types do
   @moduledoc """
   This module contains the types in Fika which are implemented using structs
   """
-
-  defmodule FunctionRef do
-    defstruct [:arg_types, :return_type]
-
-    defimpl String.Chars, for: FunctionRef do
-      def to_string(%{arg_types: arg_types, return_type: return_type}) do
-        args = Enum.join(arg_types, ",")
-        "Fn(#{args}->#{return_type})"
-      end
-    end
-  end
 end
```

Comment:
Let's delete this file?

Reply:
@emilsoman what would you rather: delete this file or add a more exaustive doc here explaining why we use this structs?

Reply:
I'd rather delete the file instead of keeping it for holding documentation alone

Reply:
Done!